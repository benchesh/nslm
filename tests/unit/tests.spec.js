const { spawn } = require('child_process');
const path = require('path');
const { path: appRoot } = require('app-root-path');
const fs = require('fs');

const pkg = require(path.resolve(appRoot,'package.json'));

async function runNslmCmd(cmd) {
  const logs = [];

  const spawnProcess = spawn('bash', ['-c',
    [
      'node',
      pkg.bin.nslm,
      cmd,
      `--homedir="${path.resolve(appRoot, 'tests')}"`
    ].join(' '),
  ], { cwd: appRoot });

  spawnProcess.stdout.on('data', (data) => {
    logs.push(data.toString());
  });

  spawnProcess.stderr.on('data', (data) => {
    logs.push(data.toString());
  });

  return new Promise((resolve) => {
    spawnProcess.on('close', async () => resolve(logs.join('').trim()));
  });
}

describe('nslm', () => {
  // describe('register', () => {

  //   it('should register the mock node modules', async () => {
  //     const test = await runNslmCmd('register');
  //     console.log(test)
  //   });
  // });

  describe('help', () => {
    it('matches the snapshot', async () => {
      const output = await runNslmCmd('help');
      expect(output).toMatchSnapshot();
    });
  });

  describe('version', () => {
    it('should return the version number', async () => {
      const output = await runNslmCmd('version');
      expect(output).toBe(`v${pkg.version}`);
    });
  });
});
