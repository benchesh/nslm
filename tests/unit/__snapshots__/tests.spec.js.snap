// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nslm help matches the snapshot 1`] = `
"Commands:
  nslm register                                                    Recursively register all node modules for use with nslm
  nslm deregister / unregister                                     Recursively deregister node modules for use with nslm
  nslm link                                                        Replace all modules for the current project with their nslm versions
  nslm delink / unlink                                             Remove all nslm modules from the current project
  nslm check-links / checklinks / check-link / checklink / check   Replace all modules for the current project with their nslm versions
  nslm help / h                                                    Show help
  nslm version / v                                                 Show version number"
`;
